### PR TITLE
3 new LED effect animations

### PR DESCRIPTION
**Pull request by TurtleHunter**
## Description
3 new LED effect animations based on subtly shifting the hue over time and then returning to the starting hue without a noticeable break.
- Hue Breathing - All LED's shift hue at the same time.
- Hue Pendulum - Hue shift changes in a wave to the right before reversing direction to the left.
- Hue Wave - Hue shift changes in a wave to the right continuously.

3 files are added, hue_breathing_anim.h; hue_pendulum_anim.h; and hue_wave_anim.h;
1 file is changed, rgb_matrix_effects.inc;
All 3 rely on the same variable name, huedelta, to determine what I like to think of as the shift in hue. Each *anim.h file has the value set statically.

Tested on a Massdrop/Alt

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by TurtleHunter** [comment]
This is my first PR ever so if I've done something wrong please advise me.

File: quantum/rgb_matrix_animations/rgb_matrix_effects.inc [hunk 1]
```diff
@@ -32,3 +32,6 @@
 #include "rgb_matrix_animations/solid_reactive_nexus.h"
 #include "rgb_matrix_animations/splash_anim.h"
 #include "rgb_matrix_animations/solid_splash_anim.h"
+#include "rgb_matrix_animations/hue_breathing_anim.h"
```
Comment (drashna) [inline quantum/rgb_matrix_animations/rgb_matrix_effects.inc:35]: Actually, could you move these to right after jellybean_raindrops (and before typing heatmap?

Comment (TurtleHunter) [reply quantum/rgb_matrix_animations/rgb_matrix_effects.inc]: Done! Thank you

File: quantum/rgb_matrix_animations/hue_breathing_anim.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    RGB rgb       = hsv_to_rgb(hsv);
+    for (uint8_t i = led_min; i < led_max; i++) {
+        RGB_MATRIX_TEST_LED_FLAGS();
+        rgb_matrix_set_color(i, rgb.r, rgb.g, rgb.b);
+    }
+    return led_max < DRIVER_LED_TOTAL;
+}
+
+#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+#endif      // DISABLE_RGB_HUE_BREATHING
```
Comment (zvecr) [inline quantum/rgb_matrix_animations/hue_breathing_anim.h:22]:
```suggestion
#ifndef DISABLE_RGB_MATRIX_HUE_BREATHING
RGB_MATRIX_EFFECT(HUE_BREATHING)
#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS

// Change huedelta to adjust range of hue change. 0-255.
// Hue Breathing - All LED's light up
bool HUE_BREATHING(effect_params_t* params) {
    RGB_MATRIX_USE_LIMITS(led_min, led_max);
    uint8_t  huedelta = 12;
    HSV      hsv      = rgb_matrix_config.hsv;
    uint16_t time     = scale16by8(g_rgb_timer, rgb_matrix_config.speed / 8);
    hsv.h             = hsv.h + scale8(abs8(sin8(time) - 128) * 2, huedelta);
    RGB rgb           = hsv_to_rgb(hsv);
    for (uint8_t i = led_min; i < led_max; i++) {
        RGB_MATRIX_TEST_LED_FLAGS();
        rgb_matrix_set_color(i, rgb.r, rgb.g, rgb.b);
    }
    return led_max < DRIVER_LED_TOTAL;
}

#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
#endif      // DISABLE_RGB_HUE_BREATHING

```

File: quantum/rgb_matrix_animations/hue_pendulum_anim.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+static HSV HUE_PENDULUM_math(HSV hsv, uint8_t i, uint8_t time) {
+	uint8_t huedelta = 12;
+    hsv.h = hsv.h + scale8(abs8(sin8(time) + (g_led_config.point[i].x) - 128) * 2, huedelta);
+    return hsv;
+}
+
+bool HUE_PENDULUM(effect_params_t* params) { return effect_runner_i(params, &HUE_PENDULUM_math); }
+
+#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+#endif      // DISABLE_RGB_HUE_PENDULUM
```
Comment (zvecr) [inline quantum/rgb_matrix_animations/hue_pendulum_anim.h:17]:
```suggestion
#ifndef DISABLE_RGB_MATRIX_HUE_PENDULUM
RGB_MATRIX_EFFECT(HUE_PENDULUM)
#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS

// Change huedelta to adjust range of hue change. 0-255.
// Looks better with a low value and slow speed for subtle change.
// Hue Pendulum - color changes in a wave to the right before reversing direction
static HSV HUE_PENDULUM_math(HSV hsv, uint8_t i, uint8_t time) {
    uint8_t huedelta = 12;
    hsv.h            = hsv.h + scale8(abs8(sin8(time) + (g_led_config.point[i].x) - 128) * 2, huedelta);
    return hsv;
}

bool HUE_PENDULUM(effect_params_t* params) { return effect_runner_i(params, &HUE_PENDULUM_math); }

#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
#endif      // DISABLE_RGB_HUE_PENDULUM

```

File: quantum/rgb_matrix_animations/hue_wave_anim.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+static HSV HUE_WAVE_math(HSV hsv, uint8_t i, uint8_t time) {
+	uint8_t huedelta = 24;
+	 hsv.h = hsv.h + scale8(abs8(g_led_config.point[i].x - time), huedelta);
+    return hsv;
+}
+
+bool HUE_WAVE(effect_params_t* params) { return effect_runner_i(params, &HUE_WAVE_math); }
+
+#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+#endif      // DISABLE_RGB_HUE_WAVE
```
Comment (zvecr) [inline quantum/rgb_matrix_animations/hue_wave_anim.h:17]:
```suggestion
#ifndef DISABLE_RGB_MATRIX_HUE_WAVE
RGB_MATRIX_EFFECT(HUE_WAVE)
#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS

// Change huedelta to adjust range of hue change. 0-255.
// Looks better with a low value and slow speed for subtle change.
// Hue Wave - color changes in a wave to the right
static HSV HUE_WAVE_math(HSV hsv, uint8_t i, uint8_t time) {
    uint8_t huedelta = 24;
    hsv.h            = hsv.h + scale8(abs8(g_led_config.point[i].x - time), huedelta);
    return hsv;
}

bool HUE_WAVE(effect_params_t* params) { return effect_runner_i(params, &HUE_WAVE_math); }

#    endif  // RGB_MATRIX_CUSTOM_EFFECT_IMPLS
#endif      // DISABLE_RGB_HUE_WAVE

```

**Comment by TurtleHunter** [comment]
Forgive me, I'm still new to Github, is there anything else I need to do? I think I committed the suggested changes

File: docs/feature_rgb_matrix.md [hunk 5]
```diff
@@ -228,6 +228,9 @@ enum rgb_matrix_effects {
     RGB_MATRIX_RAINBOW_PINWHEELS,   // Full dual gradients spinning two halfs of keyboard
     RGB_MATRIX_RAINDROPS,           // Randomly changes a single key's hue
     RGB_MATRIX_JELLYBEAN_RAINDROPS, // Randomly changes a single key's hue and saturation
+	RGB_MATRIX_HUE_BREATHING,		// Hue shifts up a slight ammount at the same time, then shifts back
+	RGB_MATRIX_HUE_PENDULUM,		// Hue shifts up a slight ammount in a wave to the right, then back to the left
+	RGB_MATRIX_HUE_WAVE,			// Hue shifts up a slight ammount and then back down in a wave to the right
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md:233]:
```suggestion
    RGB_MATRIX_HUE_BREATHING,       // Hue shifts up a slight ammount at the same time, then shifts back
    RGB_MATRIX_HUE_PENDULUM,        // Hue shifts up a slight ammount in a wave to the right, then back to the left
    RGB_MATRIX_HUE_WAVE,            // Hue shifts up a slight ammount and then back down in a wave to the right 
```

File: quantum/rgb_matrix_animations/hue_breathing_anim.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#ifndef DISABLE_RGB_MATRIX_HUE_BREATHING
+RGB_MATRIX_EFFECT(HUE_BREATHING)
+#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+
+// Change huedelta to adjust range of hue change. 0-255.
+// Hue Breathing - All LED's light up
+bool HUE_BREATHING(effect_params_t* params) {
```
Comment (fauxpark) [inline quantum/rgb_matrix_animations/hue_breathing_anim.h:7]:
```suggestion
bool HUE_BREATHING(effect_params_t *params) {

```

**Review by drashna — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**